esmvalcore/preprocessor/_shared.py

```diff
@@ -21,54 +29,154 @@ def guess_bounds(cube, coords):
 
     Parameters
     ----------
     operator:
-        A named operator.
+        A named operator that is used to search for aggregators. Will be
+        capitalized before searching for aggregators, i.e., `MEAN` **and**
+        `mean` will find :const:`iris.analysis.MEAN`.
+    **operator_kwargs:
+        Optional keyword arguments for the aggregator.
 
     Returns
     -------
-    iris.analysis.Aggregator
+    tuple[iris.analysis.Aggregator, dict]
         Object that can be used within :meth:`iris.cube.Cube.collapsed`,
         :meth:`iris.cube.Cube.aggregated_by`, or
-        :meth:`iris.cube.Cube.rolling_window`.
+        :meth:`iris.cube.Cube.rolling_window` and the corresponding keyword
+        arguments.
 
     Raises
     ------
     ValueError
-        An invalid operator is specified. Allowed options: `mean`, `median`,
-        `std_dev`, `sum`, `variance`, `min`, `max`, `rms`.
+        An invalid `operator` is specified, i.e., it is not found in
+        :mod:`iris.analysis` or the returned object is not an
+        :class:`iris.analysis.Aggregator`.
+
     """
-    operators = [
-        'mean', 'median', 'std_dev', 'sum', 'variance', 'min', 'max', 'rms',
-    ]
-    operator = operator.lower()
-    if operator not in operators:
+    cap_operator = operator.upper()
+    aggregator_kwargs = dict(operator_kwargs)
+
+    # Deprecations
+    if cap_operator == 'STD':
+        msg = (
+            f"The operator '{operator}' for computing the standard deviation "
+            f"has been deprecated in ESMValCore version 2.10.0 and is "
+            f"scheduled for removal in version 2.12.0. Please use 'std_dev' "
+            f"instead. This is an exact replacement."
+        )
+        warnings.warn(msg, ESMValCoreDeprecationWarning)
+        operator = 'std_dev'
+        cap_operator = 'STD_DEV'
+    elif re.match(r"^(P\d{1,2})(\.\d*)?$", cap_operator):
+        msg = (
+            f"Specifying percentile operators with the syntax 'pXX.YY' (here: "
+            f"'{operator}') has been deprecated in ESMValCore version 2.10.0 "
+            f"and is scheduled for removal in version 2.12.0. Please use "
+            f"`operator='percentile'` with the keyword argument "
+            "`percent='XX.YY'` instead. This is an exact replacement."
+        )
+        warnings.warn(msg, ESMValCoreDeprecationWarning)
+        aggregator_kwargs['percent'] = float(operator[1:])
+        operator = 'percentile'
+        cap_operator = 'PERCENTILE'
+
+    # Check if valid aggregator is found
+    if not hasattr(iris.analysis, cap_operator):
+        raise ValueError(
+            f"Aggregator '{operator}' not found in iris.analysis module"
+        )
+    aggregator = getattr(iris.analysis, cap_operator)
+    if not hasattr(aggregator, 'aggregate'):
+        raise ValueError(
+            f"Aggregator {aggregator} found by '{operator}' is not a valid "
+            f"iris.analysis.Aggregator"
+        )
```

Comment:
You mean a general `try ... except Exception:`? I think it's clearer the ways it is now: if the object is not an aggregator, a nice error message is raise. If something else goes wrong, this error is raised without a try/except.

Note that I need to check for the existence of the `aggregate` method here instead of a simple `isinstance(aggregator, iris.analysis.Aggregator)` because (funnily enough) not all aggregators in `iris.analysis` are actually `iris.analysis.Aggragator` objects :smile:, but rather `iris.analysis._Aggragator` (which is private, so I don't import that).

Reply:
yeah that's fine - only thing I'd be more at ease to see is the iris exception too bc exactly of this reason - am afraid they'll change the API and we'll be in the dark for a bit

Reply:
There actually is a try/except around the actual call to `aggregate` (hidden behind the `collapsed`) which will always give you the iris error

Reply:
good then, cheers, Manu! My iris-OCD is happy now :grin: